crates/protocol/interop/src/safety.rs

```diff
@@ -19,6 +21,34 @@ pub enum SafetyLevel {
     Invalid,
 }
 
+impl FromStr for SafetyLevel {
+    type Err = SafetyLevelParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "finalized" => Ok(Self::Finalized),
+            "safe" => Ok(Self::Safe),
+            "local_safe" | "localsafe" => Ok(Self::LocalSafe),
+            "cross_unsafe" | "crossunsafe" => Ok(Self::CrossUnsafe),
+            "unsafe" => Ok(Self::Unsafe),
+            "invalid" => Ok(Self::Invalid),
+            _ => Err(SafetyLevelParseError),
+        }
+    }
+}
+
+/// Error when parsing SafetyLevel from string.
+#[derive(Debug)]
+pub struct SafetyLevelParseError;
```

Comment:
Can you derive `thiserror::Error` here, rather than manually implementing `Display` + `Error`? Also, this crate is `no_std` - should not be importing any `std` crates.

Should look like:

```rs
use thiserror::Error;

// ...

/// Error when parsing SafetyLevel from string.
#[derive(Error, Debug)]
#[error("Invalid SafetyLevel")]
pub struct SafetyLevelParseError;
```